src/AppBundle/Controller/Admin/Client/ReportController.php

```diff
@@ -145,9 +145,30 @@ public function manageAction(Request $request, $id)
             return $this->render('AppBundle:Admin/Client/Report:manageConfirm.html.twig', [
                 'report' => $report,
                 'form' => $confirmForm->createView(),
+                'urlData' => [
+                    'startDate' => $form['startDate']->getData()->format('Y-m-d'),
+                    'endDate' => $form['endDate']->getData()->format('Y-m-d'),
+                    'dueDateChoice' => $form['dueDateChoice']->getData(),
+                    'dueDateCustom' => $form['dueDateCustom']->getData() !== null ? $form['dueDateCustom']->getData()->format('Y-m-d') : null,
+                    'unsubmittedSection' => $report->getUnsubmittedSectionsIds(),
+                ],
             ]);
         }
 
+        // Use URL data
+        $dataFromUrl = $request->get('data') ?: [];
+        isset($dataFromUrl['startDate']) && $form['startDate']->setData(new \DateTime($dataFromUrl['startDate']));
+        isset($dataFromUrl['endDate']) && $form['endDate']->setData(new \DateTime($dataFromUrl['endDate']));
+        isset($dataFromUrl['dueDateChoice']) && $form['dueDateChoice']->setData($dataFromUrl['dueDateChoice']);
+        isset($dataFromUrl['dueDateCustom']) && $form['dueDateCustom']->setData(new \DateTime($dataFromUrl['dueDateCustom']));
+        if (isset($dataFromUrl['unsubmittedSection'])) {
+            $unsubmittedSections = $form['unsubmittedSection']->getData();
+            foreach ($dataFromUrl['unsubmittedSection'] as $index => $name) {
+                $unsubmittedSections[$index]->setPresent(true);
```

Comment:
Are the indexes of both arrays always going to correspond to the same data?

Reply:
Almost certainly, but the code/name would be more consistent so I'll use that instead.

Reply:
@acsauk-mojd Done, no more index relying